src/display/canvas.js

```diff
@@ -1185,20 +1185,20 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
             y = args[j++];
             var width = args[j++];
             var height = args[j++];
-            if (width === 0) {
+            if (width === 0 && ctx.lineWidth < this.getSinglePixelWidth()) {
```

Comment:
Can't we do this `getSinglePixelWidth` function call only once? We're doing it four times in the worst case now, and the path construction code is usually quite hot.

Reply:
> We're doing it four times in the worst case now,

Well, that case *almost never* happens in practice since in most cases both the width *and* height is non-zero.
Hence I did it this way on purpose, since otherwise *every* single `re` operator would unnecessarily require a `this.getSinglePixelWidth()` call.

Reply:
That's seems fair; thanks!